migration/sql-files/106-number-sequences.sql

```diff
@@ -22,10 +22,13 @@ UPDATE areas SET number = seq.row_number
 FROM (SELECT id, space_id, created_at, row_number() OVER (PARTITION BY space_id ORDER BY created_at ASC) FROM areas) AS seq
 WHERE areas.id = seq.id;
 
--- Make number a required column
+-- Make number a required column and add an index for faster querying over space_id and number
 ALTER TABLE iterations ALTER COLUMN number SET NOT NULL;
 ALTER TABLE areas ALTER COLUMN number SET NOT NULL;
 
+CREATE UNIQUE INDEX iterations_space_id_number_idx ON iterations USING BTREE (space_id, number);
+CREATE UNIQUE INDEX areas_space_id_number_idx ON areas USING BTREE (space_id, number);
```

Comment:
Maybe we should alter table to add unique constraint instead of creating a unique index? Looks like this is the preferred approach for PostgreSQL: https://www.postgresql.org/docs/9.4/static/indexes-unique.html

`ALTER TABLE iterations ADD UNIQUE (space_id, number) `
or
`ALTER TABLE iterations ADD CONSTRAINT iterations_space_id UNIQUE (space_id, number);`

The doc above also states that only b-tree indexes can be declared as unique. Maybe we don't need to explicitly specify "USING BTREE" ...?

Reply:
I've changed it to the preferred syntax in b97c28e7.